js/packages/teams-ai/src/MemoryFork.ts

```diff
@@ -0,0 +1,135 @@
+
+const TEMP_SCOPE = 'temp';
+
+/**
+ * Represents a memory.
+ * @remarks
+ * A memory is a key-value store that can be used to store and retrieve values.
+ */
+export interface Memory {
+    /**
+     * Deletes a value from the memory.
+     * @param path Path to the value to delete in the form of `[scope].property`. If scope is omitted, the value is deleted from the temporary scope.
+     * @returns True if the value was deleted, false otherwise.
+     * @throws Error if the path is invalid.
+     */
+    deleteValue(path: string): void;
+
+    /**
+     * Checks if a value exists in the memory.
+     * @param path Path to the value to check in the form of `[scope].property`. If scope is omitted, the value is checked in the temporary scope.
+     * @returns True if the value exists, false otherwise.
+     */
+    hasValue(path: string): boolean;
+
+    /**
+     * Retrieves a value from the memory.
+     * @param path Path to the value to retrieve in the form of `[scope].property`. If scope is omitted, the value is retrieved from the temporary scope.
+     * @returns The value or undefined if not found.
+     */
+    getValue<TValue = unknown>(path: string): TValue;
```

Comment:
since this returns undefined when the path is not found, would it make sense to change the return type to `TValue | undefined`?